weave/artifact_wandb.py

```diff
@@ -489,7 +499,10 @@ def _set_read_artifact_uri(self, uri):
     def _saved_artifact(self):
         if self._read_artifact is None:
             uri = self._read_artifact_uri
-            path = f"{uri.entity_name}/{uri.project_name}/{uri.name}:{uri.version}"
+            if isinstance(uri, WeaveWBArtifactByIDURI):
+                path = f"{uri.path_root}/{uri.artifact_id}/{uri.name}:{uri.version}"
+            else:
+                path = f"{uri.entity_name}/{uri.project_name}/{uri.name}:{uri.version}"
             self._read_artifact = get_wandb_read_artifact(path)
```

Comment:
Maybe I am missing something, but in the case that we have `WeaveWBArtifactByIDURI`, then `path_root` will be `__wb_artifacts_by_id__` which seems like it will error?